### ISSUE TITLE
Exploring-Adversarial-Robustness-of-SSM   - tiny_imagenet results

### ISSUE THREAD

**Issue by TalRub104** [Biqing-Qi/Exploring-Adversarial-Robustness-of-Deep-State-Space-Models#8]
Hi,

I attempted to reproduce the Tiny ImageNet results presented in your paper, 'Exploring Adversarial Robustness of Deep State Space Models,' using your GitHub repository.  
In the github you referred to https://github.com/tjmoon0104/pytorch-tiny-imagenet, I  downloaded the tiny-imagenet dataset.
I am not sure if you used the prepare_dataset.py script or simply downloaded the tiny-imagenet-200 dataset.
I just downloaded the  imagenet-200 dataset using the following code:
`import os
import urllib.request
import zipfile
def download_tinyimagenet(destination="datasets"):
    dataset_dir = os.path.join(destination, "tiny-imagenet-200")
    zip_path = os.path.join(destination, "tiny-imagenet-200.zip")
    url = "http://cs231n.stanford.edu/tiny-imagenet-200.zip"

    if not os.path.exists(dataset_dir):
        os.makedirs(destination, exist_ok=True)
        print("Downloading Tiny ImageNet...")
        urllib.request.urlretrieve(url, zip_path)

        print("Extracting dataset...")
        with zipfile.ZipFile(zip_path, "r") as zip_ref:
            zip_ref.extractall(destination)

        os.remove(zip_path)
        print(f"Tiny ImageNet downloaded and extracted to {dataset_dir}")
    else:
        print("Tiny ImageNet dataset already exists.")

download_tinyimagenet()`

Then, I ran the following command:  
python train_trades_tinyimagenet.py --AT_type Nat --model_name SSM --attack_type PGD --model-dir checkpoints/model-tinyimagenet --num_layers 4 --num-classes 50 --epochs 180 --epsilon 0.031 --step-size 0.007 --beta 6

Finally, I obtained a clean accuracy of 41.76%, while you reported 20.88% in your paper (based on the last epoch checkpoints). I didn’t encounter such a large difference with CIFAR-10, so I’m wondering if the way you downloaded the dataset might explain the discrepancy.

To calculate the clean accuracy, i used the following code:

`from utils.tiny_imagenet import  build_model as tinyimagenet_build_model
from utils.tiny_imagenet import load_tinyimagenet
from utils.evalution import eval_test

train_loader, test_loader, num_classes = load_tinyimagenet(args)
model = get_model(args, ckpt_path, device)
test_loss, test_acc = eval_test(args=args, model=model, device=device, test_loader=test_loader)`

I would be grateful for any help or clarification regarding this issue.
